r3bdata/califaData/R3BCalifaMappedData.h

```diff
@@ -16,46 +16,32 @@
 #include "TObject.h"
 #include <stdint.h>
 
+class R3BCalifaFebexReader;
+
 class R3BCalifaMappedData : public TObject
 {
-
-  public:
-    // Default Constructor
-    R3BCalifaMappedData();
-
-    /** Standard Constructor
-     *@param crystalId   Crystal unique identifier
-     *@param energy      Total energy deposited in the crystal [GeV]
-     *@param nf          Total fast amplitude deposited in the crystal [a.u.]
-     *@param ns          Total slow amplitude deposited on the crystal [a.u.]
-     *@param time        Time since event start [ns]
-     *@param tot         Time over threshold
-     **/
-    R3BCalifaMappedData(UShort_t crystalId, Int_t energy, Int_t nf, Int_t ns, uint64_t time, UChar_t error, Int_t tot);
-
-    // Destructor
-    virtual ~R3BCalifaMappedData() {}
-
-    // Getters
-    inline const UShort_t& GetCrystalId() const { return fCrystalId; }
-    inline const Int_t& GetEnergy() const { return fEnergy; }
-    inline const Int_t& GetNf() const { return fNf; }
-    inline const Int_t& GetNs() const { return fNs; }
-    inline const uint64_t& GetTime() const { return fTime; }
-    inline const UChar_t& GetError() const { return fError; }
-    inline const Int_t& GetTot() const { return fTot; }
-
-  protected:
-    UShort_t fCrystalId; // crystal unique identifier
-    Int_t fEnergy;       // total energy in the crystal
-    Int_t fNf;           // total fast amplitude in the crystal
-    Int_t fNs;           // total slow amplitude in the crystal
-    uint64_t fTime;      // time-stamp (common to all the hits in the event)
-    UChar_t fError;      // bit coded error flag
-    Int_t fTot;          // time-over-treshold
-
+#define MEMBER_PLUS_GETTER(type, name)                                  \
```

Comment:
Could you please write the explicit data members and get functions in the interface?
Code is going to be written once but read by many people many times. There is no need for this additional complexity which obscure the code for many people.

Reply:
Yes, I agree with Hector.
I will adapt the code in the dev branch and check that everything works for the s455 experiment.

Reply:
I was trying to avoid the duplication between getter return types and variable types, but this is obviously better solved by N3638, e.g. auto GetFoo {return fFoo;}. Preprocessor stuff is always a crutch. 

But per [C++ core guidelines C.131](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-get), we should probably refactor this class into a struct anyhow.